docs/proposals/removal-tekton-resources.md

```diff
@@ -0,0 +1,269 @@
+
+The container images that we build must be multi-platform images.
+
+### Risks and Mitigations
+
+The complexity and the relationship with other enhancement proposals like [Add EP on Build Strategies Parametrization #697](https://github.com/shipwright-io/build/pull/697) where we add build strategy author defined parameters that - name-wise - can conflict with system parameters; with ongoing pull requests like [Remote Artifacts](https://github.com/shipwright-io/build/pull/616) that introduces the first source kind (HTTP) that we download in our own step rather than using a Tekton resource; and with future items like [Remove support for the pipeline service account #680](https://github.com/shipwright-io/build/issues/680) that I intentionally partially covered for the source secret already as I do not think it makes much sense to implement our own logic relying on something that we know will go away.
+
+To address the complexity, I suggest that the implementation is done in a staged approach with smaller but consistent changes (the numbering sometimes implies dependencies, but some things are also independent and can be worked on in parallel or in a different order):
+
+1. Introduce system parameters for the output image (`image-url` and `image-tag`), remove the image resource, use the new system parameters in the sample build strategies to access the designated location of the output image. Comment out those arguments in the Kaniko and ko strategies that write the image digest file.
+2. Implement a container image that accepts the arguments for the Git operation and runs the `git clone`, either already using [go-git](https://github.com/go-git/go-git) if that is easy to achieve, or using Tekton's approach to call the `git` binary. Tests for the binary must be implemented as well for public and private repositories.
+3. Introduce system parameters that point to the source directory (`source-root` and `source-context`) and use them in our sample build strategies as a replacement of the hardcoded `/workspace/source` paths.
+4. Introduce the `shipwright` workspace. Replace the Git resource with our own custom step that consumes the container image from (2). Stop adding the Git secret to the service account and mount it to this step directly.
+5. Add a result for the commit sha and fill it from our Git executable.
+6. Add a result for the image digest and extend the sample build strategies to write to this result.
+7. Improve the implementation for HTTP sources to make it consistent with this design.
+8. Define and implement a well-defined set of exit codes per resource kind and use them in the BuildRun controller for an enhanced error reporting.
+
+The goal to run as non-root is not covered in great detail in this proposal as some items are not 100 % clear to me yet, for example:
```

Comment:
I'd rather remove this an an explicit goal of this EP, and just mention it as a nice-to-have that this enables later on. There are a number of open questions just for this item, and I don't want to block the rest of the improvement on answering these.

Reply:
Yeah, maybe, I was trying this out today. Current state: PSP enforces MustRunAsNonRoot, git image is built with `ko` and base image ubi-minimal, I have no security context on the container nor on the pod. It works. Based on k8s docs this imo must mean that `ko` is building images that have a `USER` directive. Need to check as which user the container actually runs.